packages/usehooks-ts/src/useLocalStorage/useLocalStorage.ts

```diff
@@ -62,7 +65,11 @@ export function useLocalStorage<T>(
       setStoredValue(newValue)
 
       // We dispatch a custom event so every useLocalStorage hook are notified
-      window.dispatchEvent(new Event('local-storage'))
+      window.dispatchEvent(new CustomEvent<LocalStorageEventDetail>('local-storage', {
+        detail: {
+          key
+        }
+      }))
```

Comment:
I ran into this issue as well, and came up with this solution. This way the synthetic event matches the browser events and the handler logic doesn't need to change.
```suggestion
      const event = new Event('local-storage')
      event.key = key
      window.dispatchEvent(event)
```

Mind implementing a fix in the `useSessionStorage` hook as well? It has the same bug.